broadinstitute/single_cell_portal_core
# Log search result selection for global study search (SCP-4238)

This logs when the user selects a result from global study search.  It aims to help improve search through analytics.

Previously, when the user selected a search result, we would merely log a `click:link` event.  This had sparse information that impeded quantitative UX research on how users interact with our search interface.

Now, a much richer event is logged: `select-search-result`.  It naturally contains data on which result was selected -- including its rank in the list of search results, how many searches were done before selecting the result, and whether it was sourced from SCP or Azul / Human Cell Atlas, among other properties about the selection.  The new event also logs data about the query and larger set of results.  

The number of searches done before selection is part of click-through rate (CTR), a general search product quality indicator.

Further refinements will help.  "To dos" include making facet query data easier to analyze, and eliminating somewhat duplicative logging in both this and related pre-existing instrumentation.

To test:
* Go to home page
* Open Network panel in Chrome DevTools
* At upper-left in panel, filter requests to `bard method:POST`
* Search for a term by typing into the keyword text input
* Click on an SCP study result _and immediately press the "esc" key_
* Confirm in Network panel that a `select-search-result` event is logged to Bard / Mixpanel
* Click on an HCA / Azul result (do another keyword search if needed)
* Confirm in Network panel that a `select-search-result` event is logged
* Execute a faceted search
* Click on a result
* Confirm `select-search-result` is logged

This satisfies SCP-4238.
